### ISSUE TITLE
Looks like `desktop_photo_search/fluent_ui` is b0rked

### ISSUE THREAD

**Issue by domesticmouse** [flutter/samples#1321]
From the [CI logs](https://github.com/flutter/samples/runs/7239936799?check_suite_focus=true#step:5:279):

```console
== Testing 'desktop_photo_search/fluent_ui' on Flutter's stable channel ==
~/work/samples/samples/desktop_photo_search/fluent_ui ~/work/samples/samples
Running "flutter pub get" in fluent_ui...                           8.2s
Analyzing fluent_ui...
  error - lib/main.dart:78:7 - The element type 'dynamic' can't be assigned to the list type 'NativeSubmenu'. - list_element_type_not_assignable
  error - lib/main.dart:78:15 - The function 'Submenu' isn't defined. Try importing the library that defines 'Submenu', correcting the name to the name of an existing function, or defining a function named 'Submenu'. - undefined_function
  error - lib/main.dart:78:50 - Missing type argument for list literal. Try adding an explicit type, or remove implicit-dynamic from your analysis options file. - implicit_dynamic_list_literal
  error - lib/main.dart:79:17 - The function 'MenuItem' isn't defined. Try importing the library that defines 'MenuItem', correcting the name to the name of an existing function, or defining a function named 'MenuItem'. - undefined_function
  error - lib/main.dart:90:19 - The function 'MenuItem' isn't defined. Try importing the library that defines 'MenuItem', correcting the name to the name of an existing function, or defining a function named 'MenuItem'. - undefined_function
  error - lib/main.dart:97:7 - The element type 'dynamic' can't be assigned to the list type 'NativeSubmenu'. - list_element_type_not_assignable
  error - lib/main.dart:97:15 - The function 'Submenu' isn't defined. Try importing the library that defines 'Submenu', correcting the name to the name of an existing function, or defining a function named 'Submenu'. - undefined_function
  error - lib/main.dart:97:49 - Missing type argument for list literal. Try adding an explicit type, or remove implicit-dynamic from your analysis options file. - implicit_dynamic_list_literal
  error - lib/main.dart:98:17 - The function 'MenuItem' isn't defined. Try importing the library that defines 'MenuItem', correcting the name to the name of an existing function, or defining a function named 'MenuItem'. - undefined_function
```

**Comment by stuartmorgan** [comment]
Why isn't this using a pinned hash, so this kind of thing doesn't happen? A git reference without a hash is the equivalent of using `any` for a published package, and is a bad idea for all of the same reasons.

**Comment by domesticmouse** [comment]
ELI5 time. 

If I pin a git hash for the `menubar` and `window_size` packages, is there a process for automating the updating of the git hashes as the packages get updated?

Sure, having these samples break semi-randomly isn't great, but it is better than having the sample slowly bitrot due to a seriously out of date git hash...

**Comment by stuartmorgan** [comment]
> If I pin a git hash for the `menubar` and `window_size` packages, is there a process for automating the updating of the git hashes as the packages get updated?

I don't know if dependabot can update git references.

> but it is better than having the sample slowly bitrot due to a seriously out of date git hash...

The assumption here is that Flutter will break the plugins more often than the plugins break clients. There are absolutely no guarantees about API stability for these experimental plugins (which is why the docs for the repo explicitly say to pin a hash), so I don't think that's a safe assumption.
